src/goto-symex/slice_by_trace.cpp

```diff
@@ -219,9 +220,9 @@ void symex_slice_by_tracet::parse_events(std::string read_line)
     const std::string::size_type vnext=read_line.find(",", vidx);
     std::string event=read_line.substr(vidx, vnext - vidx);
     eis.insert(event);
-    if((!alphabet.empty()) &&
-       ((alphabet.count(event)!=0)!=alphabet_parity))
-      throw "trace uses symbol not in alphabet: "+event;
+    PRECONDITION(!alphabet.empty());
+    INVARIANT((alphabet.count(event) != 0) == alphabet_parity, 
+    	"trace uses symbol not in alphabet: " + event);
```

Comment:
Description should describe what is being asserted.... "trace symbol must be in alphabet"